option optimize_for = LITE_RUNTIME;
package gim;

import "pair.proto";

message LoginRequest{
	required string id = 1;//id of user or logic server
	optional int32 type = 2;//type -1:client, > 0:push server
	optional string token = 3;
	optional string version = 4;
	repeated Pair kvs = 5;
}


message LoginResponse{
	required int32 status = 1;//if 0 success, else fail
	optional string sessid = 2;//the session id, 
} 


message PushRequest{
	optional string from_sessid = 1;//from sessid
	optional string to_sessid = 2;//to sessid
	required string sn = 3;//each req has an unique sn
	required int32 type = 4;
	optional bytes payload = 5;// payload
}

message PushResponse{
	optional string from_sessid = 1;//from sessid
	optional string to_sessid = 2; //to sessid
	required string sn = 3;//each req has an unique sn
	required int32 status = 4;// 0 success
}

